cmake/external/firestore.cmake

```diff
@@ -18,18 +18,54 @@ if(TARGET firestore)
   return()
 endif()
 
-set(version CocoaPods-8.12.1)
-ExternalProject_Add(
-  firestore
+function(GetReleasedDep version)
+  message("Getting released firebase-ios-sdk @ ${version}")
+  ExternalProject_Add(
+    firestore
 
-  DOWNLOAD_DIR ${FIREBASE_DOWNLOAD_DIR}
-  URL https://github.com/firebase/firebase-ios-sdk/archive/${version}.tar.gz
+    DOWNLOAD_DIR ${FIREBASE_DOWNLOAD_DIR}
+    URL https://github.com/firebase/firebase-ios-sdk/archive/${version}.tar.gz
 
-  PREFIX ${PROJECT_BINARY_DIR}
+    PREFIX ${PROJECT_BINARY_DIR}
+
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    TEST_COMMAND ""
+    HTTP_HEADER "${EXTERNAL_PROJECT_HTTP_HEADER}"
+    )
+endfunction()
+
+function(GetTag t)
+  message("Getting firebase-ios-sdk from ${t}")
+  ExternalProject_Add(
+    firestore
+
+    DOWNLOAD_DIR ${FIREBASE_DOWNLOAD_DIR}
+    GIT_REPOSITORY "https://github.com/firebase/firebase-ios-sdk.git"
+    GIT_TAG ${t}
+    GIT_SHALLOW "ON"
+
+    PREFIX ${PROJECT_BINARY_DIR}
+
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND ""
+    TEST_COMMAND ""
+    HTTP_HEADER "${EXTERNAL_PROJECT_HTTP_HEADER}"
+    )
+endfunction()
+
+if(NOT FIRESTORE_DEP_SOURCE)
+  # Get from tip of the tree by default
+  GetTag("origin/master")
```

Comment:
I would really rather not get from the tip of the tree by default - this could allow our build to break, as the build tree will get out of sync with our iOS dependencies. Having the option to get from HEAD is a good idea though, just not the default.

Reply:
It certainly can break CPP because we did a commit to iOS, but IMHO this is better than having to discover this by the time we start preparing for a release. We want to catch breakages like this as earlier as possible, when the person who make the commit still have good context.

Ideally, we should be able to run the tests as part of iOS PR CI, which stops the PR from committing if it breaks. Right now the long build time of CPP is not helping with this..

Reply:
Right now, the iOS dependency version updates all at once via a workflow (example: https://github.com/firebase/firebase-cpp-sdk/pull/845). This allows us to determine at a glance (if the integration tests succeed) that the new version works. If we allow Firestore to update its version out-of-band with that, anyone's random PR could have tests fail due to the iOS SDK changing out from under them. I don't believe that is an acceptable situation.

Reply:
As discussed offline, the default behavior now remains unchanged.